diff --git a/lib/profileMapper.js b/lib/profileMapper.js
index 31ef24a..b503e97 100644
--- a/lib/profileMapper.js
+++ b/lib/profileMapper.js
@@ -1,4 +1,13 @@
 module.exports = function (raw_data) {
+  /* Mozilla LDAP specific aliases handling */
+  var all_emails = [];
+  if (raw_data.mail) {
+    all_emails = all_emails.concat(raw_data.mail);
+  }
+  if (raw_data['zimbraAlias']) {
+    all_emails = all_emails.concat(raw_data['zimbraAlias']);
+  }
+
   var profile = {
     id:          raw_data.objectGUID || raw_data.uid || raw_data.cn,
     displayName: raw_data.displayName,
@@ -8,7 +17,7 @@ module.exports = function (raw_data) {
     },
     nickname: raw_data['sAMAccountName'] || raw_data['cn'] || raw_data['commonName'],
     groups: raw_data['groups'],
-    emails: (raw_data.mail ? [{value: raw_data.mail }] : undefined)
+    emails: all_emails
   };
 
   profile['dn'] = raw_data['dn'];
@@ -29,7 +38,10 @@ module.exports = function (raw_data) {
   
   // if your LDAP service provides verified email addresses, uncomment this:
   // profile['email_verified'] = true;
-  
+
+  profile['email_verified'] = true; //LDAP emails are manually created through tickets, thus always verified
+  profile['email_aliases'] = (raw_data['zimbraAlias'] ? [{value: raw_data.mail }] : undefined)
+
   // This will make the profile huge:
   // if (raw_data['thumbnailPhoto']) {
   //   profile['picture'] = 'data:image/bmp;base64,' +
